Make `VTTRegion.lines` unsigned long type

cola119 opened the pull request:
#### dc25ee48a22ae9eea1ae639438c2ed20122f23af
<pre>
Make `VTTRegion.lines` unsigned long type
<a href="https://bugs.webkit.org/show_bug.cgi?id=261326">https://bugs.webkit.org/show_bug.cgi?id=261326</a>

Reviewed by Eric Carlson.

The type of VTTRegion.lines is changed to unsigned long in <a href="https://github.com/w3c/webvtt/issues/414">https://github.com/w3c/webvtt/issues/414</a>,
and VTTRegion.lines no longer throws IndexSizeError
when negative value was set due to the change.

* LayoutTests/imported/w3c/web-platform-tests/webvtt/api/VTTRegion/lines-expected.txt:
* LayoutTests/media/track/regions-webvtt/vtt-region-constructor-expected.txt:
* LayoutTests/media/track/regions-webvtt/vtt-region-constructor.html:
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setLines):
(WebCore::VTTRegion::parseSettingValue):
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/html/track/VTTRegion.idl:
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::scanDigits):
* Source/WebCore/html/track/VTTScanner.h:
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::collectTimeStamp):

Canonical link: <a href="https://commits.webkit.org/267827@main">https://commits.webkit.org/267827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/70ebc9d4c63aee5131bfcf5186b4d7db36f1ad10

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17705 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18035 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18571 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19530 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16561 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17900 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21325 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18184 "Built successfully") | [💥 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18625 "An unexpected error occured. Recent messages:Failed to print configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17918 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18213 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15380 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20391 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15447 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16129 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22702 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16457 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16298 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20561 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16868 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14283 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15984 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/15970 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4245 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20347 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16713 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on Source/WebCore/html/track/VTTRegion.cpp:
```diff
@@ -84,10 +84,8 @@ ExceptionOr<void> VTTRegion::setWidth(double value)
 }
 
-ExceptionOr<void> VTTRegion::setLines(int value)
+ExceptionOr<void> VTTRegion::setLines(unsigned value)
```
Can we change this function to 'void' only now that we are not throwing exception?

cola119 replied:
That makes sense. Done.

Ahmad-S792 commented on LayoutTests/media/track/regions-webvtt/vtt-region-constructor.html:
Chrome / Blink got rid of this test in favour of WPT test, if you can import relevant one, we can get rid of it. Reference: [Blink Commit](https://chromium.googlesource.com/chromium/src.git/+/ad11f82eeaba1fb9ad63c45809825d840e7ce44c)

Ahmad-S792 replied:
@cola119 - you going to delete this test in future commit? Or should I merge this?

cola119 replied:
@Ahmad-S792 I'm going to make a future PR deleting this test. Thank you.

cola119 committed:
Make `VTTRegion.lines` unsigned long type
https://bugs.webkit.org/show_bug.cgi?id=261326

Reviewed by Eric Carlson.

The type of VTTRegion.lines is changed to unsigned long in https://github.com/w3c/webvtt/issues/414,
and VTTRegion.lines no longer throws IndexSizeError
when negative value was set due to the change.

* LayoutTests/imported/w3c/web-platform-tests/webvtt/api/VTTRegion/lines-expected.txt:
* LayoutTests/media/track/regions-webvtt/vtt-region-constructor-expected.txt:
* LayoutTests/media/track/regions-webvtt/vtt-region-constructor.html:
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setLines):
(WebCore::VTTRegion::parseSettingValue):
* Source/WebCore/html/track/VTTRegion.h:
* Source/WebCore/html/track/VTTRegion.idl:
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::scanDigits):
* Source/WebCore/html/track/VTTScanner.h:
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::collectTimeStamp):

Canonical link: https://commits.webkit.org/267827@main
